FirebaseAuth/Sources/Swift/Storage/AuthSharedKeychainServices.swift

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+// TODO(ncooke3): Reshape API to conform to `AuthStorage`.
+@objc(FIRAuthSharedKeychainServices) public class AuthSharedKeychainServices: NSObject {
+  // MARK: - Private methods for shared keychain operations
+
+  /** @fn getItemWithQuery:error:
+   @brief Get the item from keychain by given query.
+   @param query The query to query the keychain.
+   @return The item of the given query. nil if not exsit.
+   */
+  @objc public func getItem(query: [String: Any]) throws -> DataWrapper {
+    var mutableQuery = query
+    mutableQuery[kSecReturnData as String] = true
+    mutableQuery[kSecReturnAttributes as String] = true
+    mutableQuery[kSecMatchLimit as String] = 2
```

Comment:
why 2?

Reply:
I'm assuming for the same reasoning here: https://github.com/firebase/firebase-ios-sdk/pull/10996/files#diff-44fc5c6e263aabae092d659983a63fc9a2f43db0468719b0dd7d7246e18a5d12L97

I'll add a similar comment.